phys/module_surface_driver.F

```diff
@@ -3863,6 +3869,212 @@ SUBROUTINE surface_driver(                                         &
+                     albedo(i,j) = ( albedo(i,j) * XICE(i,j) ) + ( (1.0-XICE(i,j)) * 0.08  )
+                     emiss(i,j) = ( emiss(i,j) * XICE(i,j) ) + ( (1.0-XICE(i,j)) * 0.98  )
+                  ENDIF
+               ENDDO
+            ENDDO
+
+            IF ( isisfc ) THEN
+               DO j=j_start(ij),j_end(ij)
+                  DO i=i_start(ij),i_end(ij)
+                     IF ( ( XICE(I,J) .GE. XICE_THRESHOLD) .AND. ( XICE(i,j) .LE. 1.0 ) ) THEN
+                        !  Weighted average of fields between ice-cover values and open-water values.
+                        flhc(i,j) = ( flhc(i,j) * XICE(i,j) ) + ( (1.0-XICE(i,j)) * flhc_sea(i,j) )
+                        flqc(i,j) = ( flqc(i,j) * XICE(i,j) ) + ( (1.0-XICE(i,j)) * flqc_sea(i,j) )
+                        cpm(i,j)  = ( cpm(i,j)  * XICE(i,j) ) + ( (1.0-XICE(i,j)) * cpm_sea(i,j)  )
+                        cqs2(i,j) = ( cqs2(i,j) * XICE(i,j) ) + ( (1.0-XICE(i,j)) * cqs2_sea(i,j) )
+                        chs2(i,j) = ( chs2(i,j) * XICE(i,j) ) + ( (1.0-XICE(i,j)) * chs2_sea(i,j) )
+                        chs(i,j)  = ( chs(i,j)  * XICE(i,j) ) + ( (1.0-XICE(i,j)) * chs_sea(i,j)  )
+                        qsfc(i,j) = ( qsfc(i,j) * XICE(i,j) ) + ( (1.0-XICE(i,j)) * qsfc_sea(i,j) )
+                        qgh(i,j)  = ( qgh(i,j)  * XICE(i,j) ) + ( (1.0-XICE(i,j)) * qgh_sea(i,j)  )
+                        qz0(i,j)  = ( qz0(i,j)  * XICE(i,j) ) + ( (1.0-XICE(i,j)) * qz0_sea(i,j)  )
+                                                            !  print *,'hfx =',hfx_sea(170,20)
+                                                            !   print *,'XICE =',XICE(170,20)
+                                                            !    print *,'QSFC =',QSFC(170,20)
+                        hfx(i,j)  = ( hfx(i,j)  * XICE(i,j) ) + ( (1.0-XICE(i,j)) * hfx_sea(i,j)  )
+                        qfx(i,j)  = ( qfx(i,j)  * XICE(i,j) ) + ( (1.0-XICE(i,j)) * qfx_sea(i,j)  )
+                        lh(i,j)   = ( lh(i,j)   * XICE(i,j) ) + ( (1.0-XICE(i,j)) * lh_sea(i,j)   )
+!save old tsk_ice
+                        tsk_save(i,j)  = tsk(i,j)
+                        tsk(i,j)  = ( tsk(i,j)  * XICE(i,j) ) + ( (1.0-XICE(i,j)) * tsk_sea(i,j)  )
+                     ENDIF
+                  ENDDO
+               ENDDO
+            ELSE
+               DO j = j_start(ij) , j_end(ij)
+                  DO i = i_start(ij) , i_end(ij)
+                     IF ( ( XICE(I,J) .GE. XICE_THRESHOLD ) .AND. ( XICE(i,j) .LE. 1.0 ) ) THEN
+                        ! Compute TSK as the open-water and ice-cover average
+                        tsk(i,j) = ( tsk(i,j) * XICE(i,j) ) + ( (1.0-XICE(i,j)) * tsk_sea(i,j) )
+                     ENDIF
+                  ENDDO
+               ENDDO
+            ENDIF
+         ENDIF
+
+      ENDIF
+
+
+
+#else
+        CALL wrf_error_fatal( "SF CTSM CURRENTLY ifdef'ed out, set -DWRF_USE_CTSM by setting&
+             & WRF_CTSM_MKFILE before running configure" )
```

Comment:
Should this be part of module_check_a_mundo.F?

Reply:
You are correct! This is not needed here, as a similar check is done in `module_check_a_mundo.F`.  I removed this conditional from phy/module_surface_driver.F. @BillSacks do you agree with this change?

Reply:
When I originally put this in (in 4e9f24e61), my feeling was that it doesn't hurt to have redundant error checks like this – and in many cases it can really help, both to catch issues as the code evolves in the future, and to make the expected behavior clear to readers. However, it looks like 2296313b introduced a bug in this logic by moving where the ifdef occurs relative to the case statement. So, as currently written, it looks like the `#else` is associated with the previous case (for CLMSCHEME), which (if I'm reading the code right) would lead to an abort whenever someone tries to run with CLMSCHEME! It looks like the same issue was introduced in `module_physics_init.F` in that commit.

So my feeling is:

- Best option: back out the changes to module_physics_init.F and module_surface_driver.F from 2296313b (unless those changes are needed for some reason?) This will mean a redundant error check, but again, my feeling is it's better to err on the side of redundancy in error checks in cases like this.

- Okay option: remove the `#else` block from both here and module_physics_init.F.

Reply:
> However, it looks like [2296313](https://github.com/wrf-model/WRF/commit/2296313bf3ff90166df5063548d05f8b1cdfe80b) introduced a bug in this logic by moving where the ifdef occurs relative to the case statement. So, as currently written, it looks like the `#else` is associated with the previous case (for CLMSCHEME), which (if I'm reading the code right) would lead to an abort whenever someone tries to run with CLMSCHEME!

Bill, I am not sure if I see correctly how does this `else` is triggered when CLMSCHEME is used. I believe this `endif` https://github.com/wrf-model/WRF/blob/9f663273504bf89e54ead7757ac00290e293520e/phys/module_surface_driver.F#L3870 is related to the CLMSCHEME 
Am I missing something?

Reply:
Anyway, I am going to remove the else block as Wei suggested. I believe this is redundant and this code block will never get triggered because this check happens in `module_check_a_mundo.F`.

https://github.com/wrf-model/WRF/blob/9f663273504bf89e54ead7757ac00290e293520e/phys/module_surface_driver.F#L4073-L4076

Reply:
Thanks, @billsacks to pointing me out to this redundant error check in`module_physics_init.F`. I removed it from there as well.

Reply:
> Bill, I am not sure if I see correctly how does this `else` is triggered when CLMSCHEME is used.

With the changes you have made, this doesn't matter any more, but to try to bring clarity to this discussion: Rereading my earlier message, I realized that I didn't state it very clearly. It's true that the preprocessor block related to CLM was ended with the earlier `#endif`. But the problem was that the `#else` block associated with WRF_USE_CTSM would have appeared under the CASE block for CLM. (I'm still not sure if I'm explaining this clearly.)